### ISSUE TITLE
Boundary segment with 3 features 

### ISSUE THREAD

**Issue by CristianMorasso** [AhmedImtiazPrio/splinecam#12]
Hello,

My name is Cristian and I'm using your tool for my project.

What I would like to do is the extraction of the boundary learned during the train of an NN.

Considering a dataset of 2 features, the tool performs very well, but when I try to increase the number of features, the output appears strange to me (even if the NN performs very well). This is an example :  

![immagine](https://github.com/AhmedImtiazPrio/splinecam/assets/61107879/5cc7c185-1e6c-4869-9260-0e7e7a49d5a7)
Desc: 3 features case; to plot the data I just apply the T matrix to transform 3 features to 2.

I've understood that the problem could be related to the T matrix, and hence to the domain.

I tried the three different functions to compute the domain (square), but I couldn't find a way to make it work.

Can you please tell me if it is possible to compute the boundary considering 3 features? Additionaly, can you tell me how can I compute the domain in the right way for the problem under analysis?

Many thanks in advance.

**Comment by AhmedImtiazPrio** [comment]
Hi Christian,

Thanks for reaching out. Indeed I think this is an issue with the
projection you are taking. Could you elaborate what domain you are using?
And to confirm you are using `get_proj_mat(domain)` to get T? If you want a
clean decision boundary between the two clusters, you'd need to consider 2D
domain along which you expect to see something close to a linear separation.

**Comment by CristianMorasso** [comment]
Thanks for your attention.
Im using this function (`get_square_slice_from_one_anchor(anchor)`) to compute the domain, and passing the mean vector as anchor, then the  `get_proj_mat(domain)` to compute the T matrix.

What you are suggesting is to use a T matrix, that performs a projection able to separate linearly the classes?
So if the data is a complex (with >3 important features), it may fail, right?

**Comment by AhmedImtiazPrio** [comment]
Yes! If the data is complex we might not see a smooth boundary between the
distributions. However, I would suggest trying with three random samples
from the two distributions with the `get_square_slice_from_centroid`
function. Might look better. If that doesn't look satisfactory, maybe try
taking the mean of the two distributions and one random sample from one of
the distributions.

**Comment by CristianMorasso** [comment]
Many thanks for your suggestions. Unfortunately, I’ve already tried in that way using a synthetic dataset, but computing the mean could be a problem when I consider two classes with two clusters for each class.

Thanks again :)
